contrib/drivers/lvm/lvm_metrics.go

```diff
@@ -127,67 +117,109 @@ func (d *MetricDriver) ValidateMetricsSupportList(metricList []string, resourceT
-		val, _ := strconv.ParseFloat(metricMap[element], 64)
-		metricValue := &model.Metric{
-			Timestamp: getCurrentUnixTimestamp(),
-			Value:     val,
+	// fill volume metrics
+	for _, volume := range volumeList {
+		convrtedVolID := convert(volume)
+		aMetricMap := metricMap[convrtedVolID]
+		aLabelMap := labelMap[convrtedVolID]
+		for _, element := range supportedMetrics {
+			val, _ := strconv.ParseFloat(aMetricMap[element], 64)
+			metricValue := &model.Metric{
+				Timestamp: getCurrentUnixTimestamp(),
+				Value:     val,
+			}
+			metricValues := make([]*model.Metric, 0)
+			metricValues = append(metricValues, metricValue)
+			metric := &model.MetricSpec{
+				InstanceID:   volume,
+				InstanceName: aMetricMap["InstanceName"],
+				Job:          "lvm",
+				Labels:       aLabelMap,
+				Component:    "volume",
+				Name:         element,
+				Unit:         metricToUnitMap[element],
+				AggrType:     "",
+				MetricValues: metricValues,
+			}
+			tempMetricArray = append(tempMetricArray, metric)
 		}
-		metricValues := make([]*model.Metric, 0)
-		metricValues = append(metricValues, metricValue)
-
-		metric := &model.MetricSpec{
-			InstanceID:   instanceID,
-			InstanceName: metricMap["InstanceName"],
-			Job:          "lvm",
-			Labels:       labelMap,
-			//Todo Take Componet from Post call, as of now it is only for volume
-			Component: "volume",
-			Name:      element,
-			//Todo : Fill units according to metric type
-			Unit: metricToUnitMap[element],
-			//Todo : Get this information dynamically ( hard coded now , as all are direct values
-			AggrType:     "",
-			MetricValues: metricValues,
+	}
+	// fill disk  metrics
+	for _, disk := range DiskList {
+		convrtedVolID := formatDiskName(disk)
+		aMetricMap := metricMap[convrtedVolID]
+		aLabelMap := labelMap[convrtedVolID]
+		for _, element := range supportedMetrics {
+			val, _ := strconv.ParseFloat(aMetricMap[element], 64)
+			metricValue := &model.Metric{
+				Timestamp: getCurrentUnixTimestamp(),
+				Value:     val,
+			}
+			metricValues := make([]*model.Metric, 0)
+			metricValues = append(metricValues, metricValue)
+			metric := &model.MetricSpec{
+				InstanceID:   disk,
+				InstanceName: aMetricMap["InstanceName"],
+				Job:          "lvm",
+				Labels:       aLabelMap,
+				Component:    "disk",
+				Name:         element,
+				Unit:         metricToUnitMap[element],
+				AggrType:     "",
+				MetricValues: metricValues,
+			}
+			tempMetricArray = append(tempMetricArray, metric)
 		}
-		tempMetricArray = append(tempMetricArray, metric)
 	}
 	metricArray := tempMetricArray
 	return metricArray, err
 }
+func convert(instanceID string) string {
+	// systat utilities (sar/iostat) returnes  volume with -- instead of -, so we need to modify volume name to map lvs output
+	instanceID = strings.Replace(instanceID, "-", "--", -1)
+	//add opensds--volumes--default-- to the start of volume
+	instanceID = "opensds--volumes--default-" + instanceID
```

Comment:
please check, default-- or default-

Reply:
Checked , Its single '-' only fin that place